test/suite/test_checkpoint37.py

```diff
@@ -0,0 +1,154 @@
+        uri = 'table:checkpoint37'
+        nrows = 1000
+
+        # Create a table.
+        ds = SimpleDataSet(
+            self, uri, 0, key_format=self.key_format, value_format=self.value_format,
+            config=self.extraconfig)
+        ds.populate()
+
+        if self.value_format == '8t':
+            value_a = 97
+            value_b = 98
+            value_c = 99
+            value_d = 100
+            value_e = 101
+        else:
+            value_a = "aaaaa" * 100
+            value_b = "bbbbb" * 100
+            value_c = "ccccc" * 100
+            value_d = "ddddd" * 100
+            value_e = "eeeee" * 100
+
+        # Write some initial data.
+        self.large_updates(ds.uri, ds, nrows, value_a, 5)
+
+        # Pin oldest and stable timestamps to 5.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(5) +
+            ',stable_timestamp=' + self.timestamp_str(5))
+
+        # Checkpoint and reopen the connection to read from the on-disk version.
+        self.session.checkpoint()
+        self.reopen_conn()
+
+        # Add updates to each key to check whether they free on reconciliation.
+        self.large_updates(ds.uri, ds, nrows, value_b, 10)
+        prev_bytes_in_use = self.get_stat(stat.conn.cache_bytes_inuse)
+        self.pr('Base bytes in use ' + str(prev_bytes_in_use))
+
+        self.large_updates(ds.uri, ds, nrows, value_c, 20)
+
+        # Pin oldest and stable timestamps to 20.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(20) +
+            ',stable_timestamp=' + self.timestamp_str(20))
+
+        # Checkpoint.
+        self.session.checkpoint()
+        bytes_in_use = self.get_stat(stat.conn.cache_bytes_inuse)
+        self.pr('After first checkpoint bytes in use ' + str(bytes_in_use))
+        self.assertLess(bytes_in_use, prev_bytes_in_use * 2)
+
+        # Another set of updates.
+        self.large_updates(ds.uri, ds, nrows, value_d, 30)
+
+        # Pin oldest and stable timestamps to 30.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(30) +
+            ',stable_timestamp=' + self.timestamp_str(30))
+
+        # Checkpoint.
+        self.session.checkpoint()
+        bytes_in_use = self.get_stat(stat.conn.cache_bytes_inuse)
+        self.pr('After second checkpoint bytes in use ' + str(bytes_in_use))
+        self.assertLess(bytes_in_use, prev_bytes_in_use * 2)
+
+        # Another set of updates.
+        self.large_updates(ds.uri, ds, nrows, value_e, 40)
+
+        # Pin oldest and stable timestamps to 40.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(40) +
+            ',stable_timestamp=' + self.timestamp_str(40))
+
+        # Checkpoint.
+        self.session.checkpoint()
+        bytes_in_use = self.get_stat(stat.conn.cache_bytes_inuse)
+        self.pr('After third checkpoint bytes in use ' + str(bytes_in_use))
+        self.assertLess(bytes_in_use, prev_bytes_in_use * 2)
```

Comment:
Should we just verify the stat cache_obsolete_updates_removed?

Reply:
Yes, added that also along with the bytes check.